homeassistant/components/anomaly/binary_sensor.py

```diff
@@ -0,0 +1,326 @@
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_FRIENDLY_NAME,
+    CONF_DEVICE_CLASS,
+    CONF_ENTITY_ID,
+    CONF_FRIENDLY_NAME,
+    CONF_SENSORS,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import State, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import generate_entity_id
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.reload import setup_reload_service
+from homeassistant.util import utcnow
+
+from . import DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ATTRIBUTE = "attribute"
+ATTR_INVERT = "invert"
+ATTR_REQUIRE_BOTH = "require_both"
+ATTR_POSITIVE_ONLY = "positive_only"
+ATTR_NEGATIVE_ONLY = "negative_only"
+ATTR_MIN_CHANGE_AMOUNT = "min_change_amount"
+ATTR_MIN_CHANGE_PERCENT = "min_change_percent"
+ATTR_SAMPLE_DURATION = "sample_duration"
+ATTR_MAX_SAMPLES = "max_samples"
+ATTR_TRAILING_SAMPLE_DURATION = "trailing_sample_duration"
+ATTR_MAX_TRAILING_SAMPLES = "max_trailing_samples"
+ATTR_SAMPLE_COUNT = "sample_count"
+ATTR_TRAILING_SAMPLE_COUNT = "trailing_sample_count"
+ATTR_SAMPLE_AVG = "sample_average"
+ATTR_TRAILING_AVG = "trailing_sample_average"
+ATTR_CHANGE_AMOUNT = "change_amount"
+ATTR_CHANGE_PERCENT = "change_percent"
+
+CONF_ATTRIBUTE = "attribute"
+CONF_INVERT = "invert"
+CONF_POSITIVE_ONLY = "positive_only"
+CONF_NEGATIVE_ONLY = "negative_only"
+CONF_REQUIRE_BOTH = "require_both"
+CONF_MAX_SAMPLES = "max_samples"
+CONF_SAMPLE_DURATION = "sample_duration"
+CONF_MAX_TRAILING_SAMPLES = "max_trailing_samples"
+CONF_TRAILING_SAMPLE_DURATION = "trailing_sample_duration"
+CONF_MIN_CHANGE_AMOUNT = "min_change_amount"
+CONF_MIN_CHANGE_PERCENT = "min_change_percent"
+
+SENSOR_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Optional(CONF_ATTRIBUTE): cv.string,
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+        vol.Optional(CONF_INVERT, default=False): cv.boolean,
+        vol.Optional(CONF_REQUIRE_BOTH, default=False): cv.boolean,
+        vol.Optional(CONF_POSITIVE_ONLY, default=False): cv.boolean,
+        vol.Optional(CONF_NEGATIVE_ONLY, default=False): cv.boolean,
+        vol.Optional(CONF_MAX_SAMPLES, default=1): cv.positive_int,
+        vol.Optional(CONF_SAMPLE_DURATION, default=0): cv.positive_int,
+        vol.Optional(CONF_MAX_TRAILING_SAMPLES, default=2): cv.positive_int,
+        vol.Optional(CONF_TRAILING_SAMPLE_DURATION, default=0): cv.positive_int,
+        vol.Optional(CONF_MIN_CHANGE_AMOUNT, default=0.0): vol.Coerce(float),
+        vol.Optional(CONF_MIN_CHANGE_PERCENT, default=0.0): vol.Coerce(float),
+    }
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_SENSORS): cv.schema_with_slug_keys(SENSOR_SCHEMA)}
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
```

Comment:
We should use our async api.
```suggestion
async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
```

Reply:
Will do.